src/notification.ts

```diff
@@ -0,0 +1,272 @@
+
+      return this.apiClient.post('/v2/notifications/sms',
+        createNotificationPayload('sms', templateId, phoneNumber, personalisation, reference, smsSenderId));
+    }
+
+    /**
+     *
+     * @param {String} notificationId
+     *
+     * @returns {Promise}
+     */
+    public getNotificationById(notificationId: string): RequestPromise {
+      return this.apiClient.get('/v2/notifications/' + notificationId);
+    }
+
+    /**
+     *
+     * @param {Object} An optional object literal that filters the query. Accepts the following filters:
+     *    @param {String} templateType
+     *    @param {String} status
+     *    @param {String} reference
+     *    @param {String} olderThanId
+     *
+     * @returns {Promise}
+     *
+     */
+    public getNotifications(options?: { templateType?: string, status?: string, reference?: string, olderThanId?: string }): RequestPromise {
+      return this.apiClient.get('/v2/notifications' + buildGetAllNotificationsQuery(options));
+    }
+
+    /**
+     *
+     * @param {String} templateId
+     *
+     * @returns {Promise}
+     */
+    public getTemplateById(templateId: string): RequestPromise {
+      return this.apiClient.get('/v2/template/' + templateId);
+    }
+
+    /**
+     *
+     * @param {String} templateId
+     * @param {Integer} version
+     *
+     * @returns {Promise}
+     */
+    public getTemplateByIdAndVersion(templateId, version): RequestPromise {
+      return this.apiClient.get('/v2/template/' + templateId + '/version/' + version);
+    }
+
+    /**
+     *
+     * @param {String} templateType
+     *
+     * @returns {Promise}
+     */
+    public getAllTemplates(templateType?: string): RequestPromise {
+        let templateQuery = '';
+
+        if (templateType) {
+          templateQuery = '?type=' + templateType;
+        }
+
+        return this.apiClient.get('/v2/templates' + templateQuery);
+    }
+
+    /**
+     *
+     * @param {String} templateId
+     * @param {Dictionary} personalisation
+     *
+     * @returns {Promise}
+     */
+    public previewTemplateById(templateId: string, personalisation?: any): RequestPromise {
+
+        let payload: any = {};
+
+        if (personalisation) {
+          payload.personalisation = personalisation;
+        }
+
+        return this.apiClient.post('/v2/template/' + templateId +  '/preview', payload);
+    }
+
+    /**
+     *
+     * @param {String} olderThanId
+     *
+     * @returns {Promise}
+     */
+    public getReceivedTexts(olderThanId?: string): RequestPromise {
+      let queryString: any = '';
+
+      if (olderThanId) {
+        queryString = '?older_than=' + olderThanId;
+      }
+
+      return this.apiClient.get('/v2/received-text-messages' + queryString);
+    }
+
+    /**
+     *
+     * @param {String} url
+     */
+    public setProxy(url: string): void {
+        this.apiClient.setProxy(url);
+    }
+}
+
+/**
+ *
+ * @param {String} type
```

Comment:
Don't care about any of these jsdocs, can delete them. Keep any actual method descriptions though.